requirements.txt

```diff
@@ -1,6 +1,16 @@
+certifi==2018.11.29
+chardet==3.0.4
 croniter==0.3.26
 Django==2.1.5
-django_compressor==2.2
+django-appconf==1.0.2
+django-compressor==2.2
+idna==2.7
 psycopg2==2.7.5
+PyMySQL==0.9.3
+python-dateutil==2.7.5
 pytz==2018.7
+rcssmin==1.0.6
 requests==2.20.1
+rjsmin==1.0.12
+six==1.12.0
+urllib3==1.24.1
```

Comment:
Is it a recommended practice to list sub-dependencies in the top-level `requirements.txt`?

Reply:
I think it's more convenient to maintain this `requirements.txt` when listing all dependencies.  
Since we could just execute `pip freeze > requirements.txt` to finish it after coding for a long time.

Reply:
I'm not convinced – what specific workflows / use case become more convenient?

A quick google search turned up this: https://medium.com/@tomagee/pip-freeze-requirements-txt-considered-harmful-f0bce66cf895